File: src/main/java/com/github/unchama/seichiassist/effect/arrow/ProjectileSpawnConfiguration.kt
package com.github.unchama.seichiassist.effect.arrow

import org.bukkit.util.Vector

data class ProjectileSpawnConfiguration(
    val speed: Double,
    private val offsetComponents: Triple<Double, Double, Double> = Triple(0.0, 0.0, 0.0),
    val gravity: Boolean = false
) {
  val offset: Vector
    get() = offsetComponents.let { (x, y, z) -> Vector(x, y, z) }
}

File: src/main/scala/com/github/unchama/seichiassist/activeskill/effect/arrow/ProjectileSpawnConfiguration.scala
package com.github.unchama.seichiassist.activeskill.effect.arrow

import org.bukkit.util.Vector

case class ProjectileSpawnConfiguration(speed: Double,
                                        private val offsetComponents: (Double, Double, Double) = (0.0, 0.0, 0.0),
                                        gravity: Boolean = false
                                       ) {
  val offset: Vector = offsetComponents match {
    case (x, y, z) => new Vector(x, y, z)
  }
}
